pkg/database/vercode.go

```diff
@@ -291,6 +291,10 @@ func (r *Realm) SaveVerificationCode(db *Database, vc *VerificationCode) error {
 	if err := vc.Validate(r); err != nil {
 		return err
 	}
+
+	// Ensure the realm is set.
+	vc.RealmID = r.ID
```

Comment:
we should probably move this up and make the Validate a BeforeSave callback

Reply:
Moved it up in https://github.com/google/exposure-notifications-verification-server/pull/2265. Moving `Validate` up will be a bit more of a refactor (but we can do it)